internal/indexcoord/meta_table.go

```diff
@@ -148,6 +149,33 @@ func (mt *metaTable) reloadMeta(indexBuildID UniqueID) (*Meta, error) {
 	return m, nil
 }
 
+func (mt *metaTable) GetNeedReleaseLockMeta(nodeIDs []UniqueID) []*indexpb.IndexMeta {
```

Comment:
move the logic out of meta table.
because meta table should not has any idea about locked, unlocked.

One thing you could do it to pass in a filter functions and find the indexmeta match certain condition